File: javascript/src/exceptions.ts
/**
 * mcp:// Protocol - Exceptions
 */

export class MCPError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'MCPError';
  }
}

export class ConnectionError extends MCPError {
  constructor(message: string) {
    super(message);
    this.name = 'ConnectionError';
  }
}

export class ToolNotFoundError extends MCPError {
  constructor(message: string) {
    super(message);
    this.name = 'ToolNotFoundError';
  }
}

export class AuthenticationError extends MCPError {
  constructor(message: string) {
    super(message);
    this.name = 'AuthenticationError';
  }
}

export class RateLimitError extends MCPError {
  constructor(message: string) {
    super(message);
    this.name = 'RateLimitError';
  }
}

export class InvalidArgumentError extends MCPError {
  constructor(message: string) {
    super(message);
    this.name = 'InvalidArgumentError';
  }
}

File: python/mcp_protocol/exceptions.py
"""
mcp:// Protocol - Exceptions
"""


class MCPError(Exception):
    """Base MCP exception"""
    pass


class ConnectionError(MCPError):
    """Connection failed"""
    pass


class ToolNotFoundError(MCPError):
    """Tool not found on server"""
    pass


class AuthenticationError(MCPError):
    """Authentication failed"""
    pass


class RateLimitError(MCPError):
    """Rate limit exceeded"""
    pass


class InvalidArgumentError(MCPError):
    """Invalid argument provided"""
    pass
